src/router/index.js

```diff
@@ -114,6 +115,14 @@ const router = new VueRouter({
   },
 });
 
+router.beforeResolve(async (to, from, next) => {
```

Comment:
Why not put this in the beforeEach?

Reply:
I put it in the `beforeResolve` rather than the `beforeEach` as the `beforeResolve` is called last before the route changes, so auth checks and everything else has been run.